src/tree/treemath.rs

```diff
@@ -38,10 +47,11 @@ pub(crate) fn node_width(n: usize) -> usize {
     }
 }
 
-pub(crate) fn root(size: LeafIndex) -> NodeIndex {
+pub(crate) fn root(size: LeafIndex) -> Result<NodeIndex, TreeMathError> {
+    check_tree_size(size)?;
```

Comment:
I don't understand what this check does. The `root()` function should never fail. It takes the number of leaves as an argument and any positive number of leaves constitutes a valid tree.

Reply:
Well, the function doesn't always produce a valid output. This should never happen but the math here doesn't know this.

Reply:
Can you give an example of invalid output?

If the output is invalid, the math should be fixed.